docs/excel/custom-functions-errors.md

```diff
@@ -1,27 +1,79 @@
 ---
-ms.date: 06/18/2019
-description: Handle errors in your Excel custom functions.
-title: Error handling for custom functions in Excel
+ms.date: 10/31/2019
+description: 'Handle and return errors like #NULL! from your custom function'
+title: Handle and return errors from your custom function (preview)
 localization_priority: Priority
 ---
 
-# Error handling within custom functions
+# Handle and return errors from your custom function (preview)
 
-When you build an add-in that defines custom functions, be sure to include error handling logic to account for runtime errors. Error handling for custom functions is the same as [error handling for the Excel JavaScript API at large](excel-add-ins-error-handling.md).
+> [!NOTE]
+> The features described in this article are currently in preview and subject to change. They are not currently supported for use in production environments. You will need to [join the Office Insider for PC](https://insider.office.com/en-us/join) to try the preview features.  A good way to try out preview features is by using an Office 365 subscription. If you don't already have an Office 365 subscription, you can get one by joining the [Office 365 Developer Program](https://developer.microsoft.com/office/dev-program).
 
-[!include[Excel custom functions note](../includes/excel-custom-functions-note.md)]
+If something goes wrong while your custom function runs, you will need to return an error to inform the user. If you have specific parameter requirements, such as only positive numbers, you will need to test the parameters and throw an error if they are not correct. You can also use a try catch block to catch any errors that occur while your custom function runs.
 
-In the following code sample, `.catch` will handle any errors that occur previously in the code.
+## Detect and throw an error
 
-```js
+Let’s look at a case where you need to ensure that a zip code parameter is in the correct format for the custom function to work. The following custom function uses a regular expression to check the zip code. If it is correct, then it will look up the city (in another function) and return the value. If it is not correct, it returns a #VALUE! error to the cell.
+
+```typescript
+/**
+* Gets a city name for the given zip code.
+* @customfunction
+* @param {string} zipCode
+* @returns The city of the zip code.
+*/
+function getCity(zipCode: string): string {
+  let isValidZip = /(^\d{5}$)|(^\d{5}-\d{4}$)/.test(zipCode);
+  if (isValidZip) return cityLookup(zipCode);
+  let error = new CustomFunctions.Error(CustomFunctions.ErrorCode.invalidValue, "Please provide a valid zip code.");
+  throw error;
+}
+```
+
+## The CustomFunctions.Error object
+
+The **CustomFunctions.Error** object is used to return an error back to the cell. When you create the object, specify which error you want to use by using one of the **ErrorCode** enum values. The following **ErrorCode** enum values can be used.
```

Comment:
Object|function|property names should be formatted as code, to prevent localization.